core/src/main/java/com/linecorp/armeria/internal/common/ArmeriaHttpUtil.java

```diff
@@ -1009,17 +1028,54 @@ public static ResponseHeaders setOrRemoveContentLength(ResponseHeaders headers,
             return headers;
         }
 
-        if (!headers.contains(HttpHeaderNames.CONTENT_LENGTH) || !content.isEmpty()) {
+        final long contentLength;
+        final boolean streamingContent = headers.isContentLengthSet() && headers.contentLength() == -1;
+        if (streamingContent) {
```

Comment:
So we treat this response as a streaming response which has one chunk, is that correct?

Reply:
The size of chunks is nondeterministic. It depends on how users set the content.